homeassistant/helpers/service.py

```diff
@@ -672,13 +690,16 @@ def async_set_service_schema(
 async def entity_service_call(  # noqa: C901
     hass: HomeAssistant,
     platforms: Iterable[EntityPlatform],
-    func: str | Callable[..., Any],
+    func: str | Callable[..., Coroutine[Any, Any, ServiceResult] | None],
     call: ServiceCall,
     required_features: Iterable[int] | None = None,
-) -> None:
+) -> ServiceResult:
```

Comment:
I feel like we could choose to not update this method at this time? That way we can avoid the discussion how to return the results.

Reply:
My impression is that services.async_call calls the method that is registered with hass.services.async_register. The entity platform and entity component async_register_entity_service register a callback that calls entity_service_call, so this seems to be used anytime services are registered that way. My impression is that we need this method in order to interoperate with services registered that way. Am i right that we want to support entity services or is that something we don't want, or perhaps I am missing another aspect of this.

Reply:
Your analysis is correct.

Unless we don't want to support entity services now, I think we need to figure out what to do for entity service calls for multiple entities etc to keep the user interface intuitive and consistent. I don't think we should recommend registering entity services directly without this helper.

We could say that we don't support service results for more than a single entity at a time and raise an exception.

Reply:
I agree this would need more thought, and it might be a good idea to keep it out of the first iteration.

Solutions could include, for example, returning a dict[str, ServiceResult], in which the str is the entity_id it applies too.

That said, a non-entity service call, could be made against one or more target entities as well (using the selector).

Reply:
OK, great, I'll revert the changes here and we can address incrementally.

The actual initial use case I had in mind was for the calendar entity, which I assumed made sense to use the entity component for, like its other services. Either way, i'll drop from this PR given it has many other interesting aspects to focus on and we can work through in parallel with other topics.